src/cache.cc

```diff
@@ -620,16 +618,16 @@ void CACHE::finish_packet(const response_type& packet)
   }
 
   // MSHR holds the most updated information about this request
-  mshr_type::returned_value finished_value{packet.data, packet.pf_metadata};
+  fill_type::returned_value finished_value{packet.data, packet.pf_metadata};
   mshr_entry->data_promise = champsim::waitable{finished_value, current_time + (warmup ? champsim::chrono::clock::duration{} : FILL_LATENCY)};
   if constexpr (champsim::debug_print) {
     fmt::print("[{}_MSHR] finish_packet instr_id: {} address: {} data: {} type: {} current: {}\n", this->NAME, mshr_entry->instr_id, mshr_entry->address,
                mshr_entry->data_promise->data, access_type_names.at(champsim::to_underlying(mshr_entry->type)), current_time.time_since_epoch() / clock_period);
   }
 
-  // Order this entry after previously-returned entries, but before non-returned
-  // entries
-  std::iter_swap(mshr_entry, first_unreturned);
+  std::iter_swap(mshr_entry, std::begin(MSHR));
+  inflight_fills.push_back(MSHR.front());
+  MSHR.pop_front();
```

Comment:
```suggestion
  inflight_fills.push_back(*mshr_entry);
  MSHR.erase(mshr_entry);
```
Does it still make sense given your changes to swap the entry to the front, then pop it?

Reply:
Yes, swapping the entry to the front and popping it is often faster than directly removing the entry.
`MSHR` is `std::deque`, and popping the front only requires O(1) while directly removing an entry requires O(n).